Ai commandline analysis playbook

curtismidd opened the pull request:
Required items, please complete
   
   Change(s):
   - Added a new playbook to perform AI Commandline analysis using GPT-3

   Reason for Change(s):
   - Usage of GPT-3 to enrich Commandline entities.

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

curtismidd commented:
@microsoft-github-policy-service agree

rahul0216 commented on Playbooks/AS-AI-Commandline-Analysis/azuredeploy.json:
```diff
@@ -0,0 +1,288 @@
+        },
+        "userprincipalname":{
+            "defaultValue": "user@domain",
+            "type": "string"
```
1. Can you use PascalCase for parameter names? Like PlaybookName instead of  playbook_name. It renders better. 
If you update the parameter name, don't forget to update wherever it is being used. 
2. You can provide default value for playbook name parameter as well.

curtismidd replied:
Thanks, pascal case issue resolved in 40730a1.

The 'userprincipalname' parameter has also been removed as managed identity is now used for this connection.

rahul0216 commented on Playbooks/AS-AI-Commandline-Analysis/azuredeploy.json:
```diff
@@ -0,0 +1,288 @@
+                "customParameterValues": {},
+                "api": {
+                    "id": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Web/locations/', resourceGroup().location,'/managedApis/azuresentinel')]"
+                }
```
I would suggest using Managed Identity for Sentinel connection. It will remove the requirement for user principal name and experience is more user friendly.  Below screenshot suggests what you need to add to enabled MI based authentication. Taken from this playbook, https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Neustar%20IP%20GeoPoint/Playbooks/Neustar-GetIPGeoInfo/azuredeploy.json
![delete](https://user-images.githubusercontent.com/8243910/216214740-08f63569-19da-43a9-8007-5f5d50a261ec.jpg)

curtismidd replied:
Thanks for the feedback on this. As I am working on the changes I have also observed the following has to be added to make the managed identity work by setting the identity of the playbook to system assigned.
![identityfield](https://user-images.githubusercontent.com/45267272/216281410-817648d0-9d9d-4195-9466-6ec2536db298.png)

curtismidd replied:
Resolved in 40730a1

rahul0216 reviewed (commented):
1. Please address the inline comments.
2. These two conditions look identical to me, or am I missing something?
![image](https://user-images.githubusercontent.com/8243910/216216523-3231561f-2a31-4272-8462-fac46ef9a3b4.png)

curtismidd commented on Playbooks/AS-AI-Commandline-Analysis/readme.md:
```diff
@@ -0,0 +1,55 @@
+Ensure you select a Playbook name
+
+Enter the username that you will use for the Microsoft Sentinel Connector
+#
```
Thanks, resolved in 
40730a1.

curtismidd commented:
> 1. Please address the inline comments.

I have made changes based on the feedback, please let me know if there is anything else needed.

Other changes made during this:

1.  Changed the API connection name from azuresentinel-PlaybookName to MicrosoftSentinel-PlaybookName
2. Added instructions for adding the responder role due to the use of a managed identity now. 

I look forward to any other feedback you may have.

rahul0216 reviewed (approved):
Changes look good, Thanks.
Can you share screenshot of a successful run?

curtismidd commented:
Thanks @rahul0216, here is a screenshot of a complete run.

![completerun](https://user-images.githubusercontent.com/45267272/216541467-d20a324c-b8f1-4c6d-a2c4-c04732c7d28e.png)

curtismidd commented:
Hi @v-atulyadav, my changes have been approved by the reviewer. Is anything further needed from my side?

v-rucdu reviewed (approved):
Approving to merge the PR
